gluten-core/src/main/scala/io/glutenproject/utils/QueryPlanSelector.scala

```diff
@@ -43,12 +43,20 @@ abstract class QueryPlanSelector[T <: QueryPlan[_]] extends Logging {
     new Throwable().getStackTrace().slice(trim, trim + max).mkString("\n")
   }
 
+  private def isGlutenStatementEnable(session: SparkSession): Boolean = {
+    if (session.sparkContext.getLocalProperty(GlutenConfig.GLUTEN_STATEMENT_ENABLE_KEY) != null) {
+      session.sparkContext.getLocalProperty(GlutenConfig.GLUTEN_STATEMENT_ENABLE_KEY).toBoolean
+    } else {
+      true
+    }
+  }
+
   protected def validate(plan: T): Boolean
 
   private[this] def shouldUseGluten(session: SparkSession, plan: T): Boolean = {
     val glutenEnabled = session.conf
       .get(GlutenConfig.GLUTEN_ENABLE_KEY, GlutenConfig.GLUTEN_ENABLE_BY_DEFAULT.toString)
-      .toBoolean
+      .toBoolean & isGlutenStatementEnable(session)
```

Comment:
better to use `&&`, so if the firsst condition if false, the second condition code will be skipped